### ISSUE TITLE
got a flat  line using CoxPHSurvivalAnalysis().predict_survival_function

### ISSUE THREAD

**Issue by solren** [sebp/scikit-survival#57]
Hi when I try to run the similar code in the example using my own data
pred_surv = estimator.predict_survival_function(x_new)#predict_survival_function(x_new)
for i, c in enumerate(pred_surv):
    plt.step(c.x, c.y, where="post", label="Sample %d" % (i + 1))
plt.ylabel("est. probability of survival $\hat{S}(t)$")
plt.xlabel("time $t$")
plt.legend(loc="best")

I got a flat line.
but the estimator score give the correct numbers:
estimator.score(data_x_numeric, data_y)
0.6753335594472886

Do you have any clue?

**Comment by sebp** [comment]
Can you please provide a small example so I could try to reproduce the problem you are having?

**Comment by sebp** [comment]
I'm closing this issue as there has been no further progress.

**Comment by LiuCMU** [comment]
hello, thanks for the great packagge!

I met the same problem, do you have any clue about this phenomenon?

```
from sksurv.linear_model import CoxPHSurvivalAnalysis

estimator = CoxPHSurvivalAnalysis()
Y_train2.columns = ['event', 'time']
Y_train2_array = Y_train2.to_records(index=False)
estimator.fit(X_train.astype(float), Y_train2_array)

pred_surv = estimator.predict_survival_function(X_train.iloc[:5, :])
time_points = np.arange(1, 38)
for i, surv_func in enumerate(pred_surv):
    plt.step(time_points, surv_func(time_points), where="post",
             label="Sample %d" % (i + 1))
plt.ylabel("est. probability of survival $\hat{S}(t)$")
plt.xlabel("time $t$")
plt.legend(loc="best")
```
![image](https://github.com/sebp/scikit-survival/assets/60156077/853fa4ec-1ee0-43b5-8c36-488f2d3d22e7)

You can see that there are 37 time steps,but only 3 steps were plotted here. This is because the following step values are predicted to be inf by the `surv_func`. 

Although the above plot is not desirable, the ROC score is very high:
```
estimator.score(X_train.iloc[:100000], Y_train_array[:100000])  ## the output is 0.99
```

Unfortunately, the training data set contains ~4millon data points, so I can't  attach here.

**Comment by cpoerschke** [comment]
Hi @LiuCMU!

I'm a new-ish package user here but enjoy learning and from having looked a little into the #249 issue wanted to share some thoughts here in case they might be helpful, perhaps some of them you've tried out already ...

* Which version of the package is used? Assuming v0.20.0 version -- https://github.com/sebp/scikit-survival/blob/v0.20.0/sksurv/linear_model/coxph.py#L566 -- if `return_array=True` is passed, is the flatness happening then too?

* `np.arange(1, 38)` are the `time_points` used and after step 3 `inf` is predicted - how does that relate to the time points in the training data e.g. when is the first/next/last censored or uncensored time point in the training data?
 
* You mention ~4million data points in the set, is it possible to reduce to 2m to 1m to 500k etc. etc. whilst still observing the flatness?

* Locally adding some logging in the `StepFunction` -- https://github.com/sebp/scikit-survival/blob/v0.20.0/sksurv/functions.py#L20-L74 -- might be interesting, you mention `inf` being predicted i.e. where/how is that happening might be gleaned from that, perhaps.

Hope this helps.

**Comment by LiuCMU** [comment]
Hi @cpoerschke 

Thanks for the suggestions! I moved on without using it because the project due a week ago. I will try these tips next time I have similar issues. Thank you very much! I appreciate it.

**Comment by sebp** [comment]
@LiuCMU It's difficult to give good advice, if I don't have access to the data, but you could try to use sklearn's StandardScaler to per-proceess your features.
